GoRogue/MapGeneration/RegionWithComponents.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections;
+using System.Collections.Generic;
+using GoRogue.Components;
+using GoRogue.Components.ParentAware;
+using SadRogue.Primitives;
+
+namespace GoRogue.MapGeneration
+{
+    /// <summary>
+    /// A Class which contains a Region, and a Component Collection.
+    /// </summary>
+    public class RegionWithComponents : IObjectWithComponents, IReadOnlyArea
```

Comment:
General note: recommend parameter consistency with the `Region` type on functions that forward to your `Region` property.